ui-tests/src/test/java/com/sonymobile/jenkins/plugins/kerberossso/KerberosSsoTest.java

```diff
@@ -228,13 +233,104 @@ private void assertAnonymousWithoutCredentials() throws IOException {
         assertThat(out, containsString("Anonymous"));
     }
 
-    private FirefoxDriver getNegotiatingFirefox(KerberosContainer kdc, String tokenCache) {
-        FirefoxOptions firefoxOptions = new FirefoxOptions();
-        FirefoxProfile profile = new FirefoxProfile();
-        profile.setAlwaysLoadNoFocusLib(true);
+    private WebDriver getNegotiatingFirefox(KerberosContainer kdc) throws IOException {
+        final String containerName = "selenium container for negotiation";
+        final String image = "selenium/standalone-firefox-debug:3.14.0";
```

Comment:
I wouldn't expect the test dealing with such ATH internals at this level. If I understood correctly https://github.com/jenkinsci/acceptance-test-harness/pull/556 would be dealing with this. Am I not understanding something?

Reply:
You are correct. This is a special case when the test needs to launch a dedicated browser context as the GSS negotiation integrates with the browser itself. It is needed to pass envvars and custom profile to the browser, which requires flexibility ATH does not provide (and otherwise have no need to provide). So this is hackish and nasty, I agree, but I prefer that over adding piles of extensibility into ATH for a single legitimate client.